src/python/pants/backend/docker/goals/tailor.py

```diff
@@ -36,7 +36,11 @@ async def find_putative_targets(
         dirname, filename = os.path.split(dockerfile)
         pts.append(
             PutativeTarget.for_target_type(
-                DockerImageTarget, dirname, "docker", [filename], kwargs={"name": "docker"}
+                DockerImageTarget,
+                dirname,
+                "docker",
+                [],
+                kwargs={"name": "docker", "source": filename},
```

Comment:
What's your thinking for whether to have `tailor` set `source=` by default? We usually never tailor default field values so that BUILD files are smaller.

Reply:
Good catch. Should probably remove that. Been so many turns back and forth so I lost track of this :p

Reply:
Oh, right, now I remember. Tailor does not yet support targets with `SingleSourceField` kind of sources.
So if I put the filename in the brackets, I hit this error: 
https://github.com/pantsbuild/pants/blob/5798377b2876dc1d5355e407972a441ea7af06d7/src/python/pants/core/goals/tailor.py#L133-L137

Reply:
But here, we could be smart, and look at the filename, and leave it out if it matches the default.

Reply:
I registered this issue https://github.com/pantsbuild/pants/issues/13600 to track this.